sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -289,10 +302,20 @@ bool Scheduler::removeMemoryObject(detail::SYCLMemObjI *MemObj,
     waitForRecordToFinish(Record, Lock);
   }
   {
+    // If allowWait is false, it means the application is shutting down.
+    // On Windows we can't safely wait on threads, because they have likely been
+    // abandoned. So we will try to get the lock. If we can, great, we'll remove
+    // the record. But if we can't, we just skip. The OS will reclaim the
+    // memory.
     WriteLockT Lock = StrictLock ? acquireWriteLock()
                                  : WriteLockT(MGraphLock, std::try_to_lock);
-    if (!Lock.owns_lock())
-      return false;
+    if (!Lock.owns_lock()) {
+
+      if (allowWait)
+        return false; // Record was not removed, the caller may try again.
+      else
+        return true; // skip.
+    }
```

Comment:
Nit;
```suggestion
    // If we fail to acquire the lock, return false if the record is still there
    // and the caller can try again. Return true if the record is gone and
    // the caller should skip.
    if (!Lock.owns_lock())
      return !allowWait;
```

Reply:
Since you have said this is only a "nit" I'd like to keep it as is, as I think it is easier to read and see the logic as it's written now. Returning `!allowWait` is definitely an abbreviation, but it doesn't tie into the comment.

Reply:
Personally, I don't think `allowWait` has any clear ties either way. In general, `if (X) return false else return true` is a bit of a logics anti-pattern. You can leave it, but as an outside reader, I think they convey the information equally well. You can also reword the comment if you can think of a better way of explaining it.